### ISSUE TITLE
Property in nested schema not saved

### ISSUE THREAD

**Issue by davidolivier1337** [Automattic/mongoose#5406]
I don't know if it's a bug or a bad use of mongoose schemas.

But, if I don't specified the property to update with 'markModified', the new value added to the nested schema is not saved.

```
var nestedValueSchema = new mongoose.Schema({
    _id: false,
    value: Number
  });
  var nestedPropertySchema = new mongoose.Schema({
    _id: false,
    active: Boolean,
    nestedValue: nestedValueSchema
  });
  var nestedSchema = new mongoose.Schema({
    _id: false,
    nestedProperty: nestedPropertySchema,
    nestedTwoProperty: nestedPropertySchema
  });
  var optionsSchema = new mongoose.Schema({
    _id: false,
    nestedField: nestedSchema
  });
  var TestSchema = new mongoose.Schema({
    fieldOne: String,
    options: optionsSchema
  });

  var Test = mongoose.model('Test', TestSchema);

  var test = new Test({
    fieldOne: 'Test One',
    options: {
      nestedField: {
        nestedProperty: {
          active: true,
          nestedValue: {
            value: 42
          }
        }
      }
    }
  });
  
  return test
    .save()
    .then(function () {
      test.options.nestedField.nestedTwoProperty = {
        active: true,
        nestedValue: {
          value: 1337
        }
      };
      // test.markModified('options');
      return test
        .save();
    })
    .then(function () {
      return Test.findOne({_id: test._id})
        .then(function (data) {
          console.log(data.options.nestedField);
        });
    });
```

So, If I un-comment 'test.markModified('options');', the 'test.options.nestedField.nestedTwoProperty' is saved.

**Without test.markModified('options'):**
`{ nestedProperty: { active: true, nestedValue: { value: 42 } } }`

**With test.markModified('options'):**
`{ nestedTwoProperty: { active: true, nestedValue: { value: 1337 } },
  nestedProperty: { active: true, nestedValue: { value: 42 } } }`

**Comment by varunjayaraman** [comment]
yeah this looks like a problem with change detection related to `.save`. If I retrieve a completely separate document instead of mutating the document being saved, it works:

```javascript
const mongoose = require('mongoose');
const co = require('co');
mongoose.Promise = global.Promise;
const GITHUB_ISSUE = `gh-5406`;


exec()
  .then(() => {
    console.log('successfully ran program');
    process.exit(0);
  })
  .catch(error => {
    console.error(`Error: ${error}\n${error.stack}`);
  });


function exec() {
  return co(function* () {
    const db = mongoose.createConnection(`mongodb://localhost:27017/${GITHUB_ISSUE}`)
    var nestedValueSchema = new mongoose.Schema({
      _id: false,
      value: Number
    });
    var nestedPropertySchema = new mongoose.Schema({
      _id: false,
      active: Boolean,
      nestedValue: nestedValueSchema
    });
    var nestedSchema = new mongoose.Schema({
      _id: false,
      nestedProperty: nestedPropertySchema,
      nestedTwoProperty: nestedPropertySchema
    });
    var optionsSchema = new mongoose.Schema({
      _id: false,
      nestedField: nestedSchema
    });
    var TestSchema = new mongoose.Schema({
      fieldOne: String,
      options: optionsSchema
    });

    var Test = db.model('Test', TestSchema);

    var test = new Test({
      fieldOne: 'Test One',
      options: {
        nestedField: {
          nestedProperty: {
            active: true,
            nestedValue: {
              value: 42
            }
          }
        }
      }
    });

    yield test.save();

    const update = yield Test.findById(test._id);
    update.options.nestedField.nestedTwoProperty = {
      active: true,
      nestedValue: {
        value: 1337
      }
    };

    yield update.save();

    const final = yield Test.findById(update._id);
    console.log(final.options.nestedField);
    // return test
    //   .save()
    //   .then(function() {
    //     test.options.nestedField.nestedTwoProperty = {
    //       active: true,
    //       nestedValue: {
    //         value: 1337
    //       }
    //     };
    //     test.markModified('options');
    //     return test
    //       .save();
    //   })
    //   .then(function() {
    //     return Test.findOne({ _id: test._id })
    //       .then(function(data) {
    //         console.log(data.options.nestedField);
    //       });
    //   });
  });
}
``

But this wont:

```javascript
const mongoose = require('mongoose');
const co = require('co');
mongoose.Promise = global.Promise;
const GITHUB_ISSUE = `gh-5406`;


exec()
  .then(() => {
    console.log('successfully ran program');
    process.exit(0);
  })
  .catch(error => {
    console.error(`Error: ${error}\n${error.stack}`);
  });


function exec() {
  return co(function* () {
    const db = mongoose.createConnection(`mongodb://localhost:27017/${GITHUB_ISSUE}`)
    var nestedValueSchema = new mongoose.Schema({
      _id: false,
      value: Number
    });
    var nestedPropertySchema = new mongoose.Schema({
      _id: false,
      active: Boolean,
      nestedValue: nestedValueSchema
    });
    var nestedSchema = new mongoose.Schema({
      _id: false,
      nestedProperty: nestedPropertySchema,
      nestedTwoProperty: nestedPropertySchema
    });
    var optionsSchema = new mongoose.Schema({
      _id: false,
      nestedField: nestedSchema
    });
    var TestSchema = new mongoose.Schema({
      fieldOne: String,
      options: optionsSchema
    });

    var Test = db.model('Test', TestSchema);

    var test = new Test({
      fieldOne: 'Test One',
      options: {
        nestedField: {
          nestedProperty: {
            active: true,
            nestedValue: {
              value: 42
            }
          }
        }
      }
    });


    return test
      .save()
      .then(function() {
        test.options.nestedField.nestedTwoProperty = {
          active: true,
          nestedValue: {
            value: 1337
          }
        };
        // test.markModified('options');
        return test
          .save();
      })
      .then(function() {
        return Test.findOne({ _id: test._id })
          .then(function(data) {
            console.log(data.options.nestedField);
          });
      });
  });
}
```

**Comment by vkarpov15** [comment]
Fix will be released in 4.11.2 :+1:
